typing/typedtree.ml

```diff
@@ -845,3 +845,19 @@ let split_pattern pat =
         combine_opts (into cpat) exns1 exns2
   in
   split_pattern pat
+
+(* Expressions are considered short if they satisfy one of:
+
+   - Looking like an identifier, small words separated with a '.' or a '#'.
+   - Do not contain spaces when printed.
+   - Look nice as the subject of a sentence, eg. "'Foo.bar' is short." *)
+let rec exp_is_short exp =
```

Comment:
The name `is_short` gives the impression that it is the size of the expression that matters but we consider that `Tell.Me.O.Muse.Of.That.Ingenious.Hero.Who.Travelled.Far.And.wide` is short. I would rather use your nice description `Look nice as the subject of a sentence`, which could be summarized as `exp_is_nominal`.